app/assets/stylesheets/layout/_top_menu.sass

```diff
@@ -181,6 +154,55 @@
     table td
       padding: 3px
 
+.top-menu-search
+  display: inline-block
+  &:not(.-collapsed) > div
+    position: relative
+    padding-top: 9px
+
+.top-menu-search--wrapper
+  float: left
+
+.top-menu-search--button
+  line-height: $header-height
+  height: $header-height
+  @include default-font($header-item-font-color, $header-item-font-size)
+  background: transparent
+  &:hover
+    text-decoration: none
+
+  .top-menu-search:not(.-collapsed) &
```

Comment:
I agree and will follow the style guide Alex linked before.
For the single-state toggle, assuming expanded is the default rule forces me to either use this style or override several values in the collapsed state (which I would rather avoid). Do you have any concern regarding the `:not` rule?